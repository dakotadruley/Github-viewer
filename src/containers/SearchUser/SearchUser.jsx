import React, { Component } from 'react';
import Username from '../../components/Username/Username.jsx';
import UserDisplay from '../../components/UserDisplay/UserDisplay.jsx';
import { fetchUser } from '../../services/fetchUser.js';

export default class SearchUser extends Component {
  state = { 
    username: 'dakotadruley',
    userdisplay: {},
    loading: true };
    
    handleClick = ({ target }) => {
      this.setState({ [target.name]: target.value });

    };
  
    componentDidMount() {
      fetchUser(this.state.username)
        .then(userdisplay => this.setState({ userdisplay, loading: false }));
    }
    
    render() {
      const { username, userdisplay, loading } = this.state;
      if(loading) return (<h1>LOADIN</h1>);
      
      return (
        <>
          <Username username={username}
            onClick={this.handleClick} />
          <UserDisplay {...userdisplay} />
        </>
      );
    }
}
