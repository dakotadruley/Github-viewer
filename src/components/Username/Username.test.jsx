import React from 'react';
import { shallow } from 'enzyme';
import SearchUser from './Username.jsx';

describe('SearchUser', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<SearchUser 
      username="dakotadruley"
      onClick={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
