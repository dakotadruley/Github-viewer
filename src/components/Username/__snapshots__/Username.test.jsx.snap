// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Username matches a snapshot 1`] = `
<Fragment>
  <input
    name="username"
    type="text"
    value="dakotadruley"
  />
  <button
    name="username"
    onClick={[Function]}
    type="button"
    value="dakotadruley"
  >
    Search
  </button>
</Fragment>
`;
